drivers/mm/mm_drv_common.c

```diff
@@ -310,38 +387,46 @@ int sys_mm_drv_simple_move_region(void *virt_old, size_t size,
 		uint8_t *va_new = (uint8_t *)virt_new + offset;
 		uintptr_t pa = phys_new + offset;
 		uint32_t flags;
-		int ret2;
```

Comment:
can this function not simply be implemented as
```
map(new)
memcpy(new, old)
unmap(old)
```
? Or do we really need to support different flags per page?

Reply:
It is more about preserving existing flags.